usb-power-profiling.js

```diff
@@ -1248,6 +1246,47 @@ function profileFromData() {
   return profile;
 }
 
+function getPowerData(start, end) {
+  let counters = [];
+  for (let device of gDevices) {
+    const {samples, sampleTimes, deviceName} = device;
+
+    let timeStart = parseFloat(start) - startTime;
+    let startIndex = 0;
+    while (sampleTimes[startIndex] < timeStart) {
+      ++startIndex;
+    }
+
+    let timeEnd = parseFloat(end) - startTime;
+    if (timeEnd < 0) {
+      throw "The requested end time is before this instance of the script was started."
```

Comment:
I was initially confused by this and almost wrote a comment saying we should continue instead of throwing as we are in a loop, then I realized this is validating the parameters given to the function, which will be the same for all iterations of the loop anyway. So to avoid this confusion, I would prefer if the lines related to `timeStart` and `timeEnd` were moved outside of the loop.